src/AI/WallyAI.java

```diff
@@ -697,6 +699,11 @@ private GameAction evictUnit(
       log(String.format("%s  Eviction cycle! Bailing.", spacing));
       return null;
     }
+    if( evictionStack.size() > EVICTION_STACK_MAX_DEPTH )
+    {
+      log(String.format("%s  Too many units blocking! Bailing.", spacing));
```

Comment:
Does this actually ever become a problem in non-recursive cases (I know you also check recursion separately)?

Reply:
Pretty sure It's a significant speedup in big unit clogs. Which Wally v Wally *regularly* produces, lol